tests/testRequest.py

```diff
@@ -653,14 +653,13 @@ def onfinish(r, result):
 
         req = Request(f)
         req.notify_finished(partial(onfinish, req))
-
         req.submit()
         req.wait()
         del req
 
         # The ThreadPool._Worker loop has a local reference (next_task),
         # so wait just a tic for the ThreadPool worker to cycle back to the top of its loop (and discard the reference)
-        time.sleep(0.1)
+        gc.collect()
```

Comment:
Do we really need this in a test?

Reply:
It broke when I didn't free the reference :) More robust test would probably try to overload thread pool with work and than check if there was any memory leakage.